src/Microsoft.Azure.SignalR.AspNet/EndpointProvider/ServiceEndpointProvider.cs

```diff
@@ -53,13 +54,14 @@ private string GetPrefixedHubName(string applicationName, string hubName)
             return string.IsNullOrEmpty(applicationName) ? hubName.ToLower() : $"{applicationName.ToLower()}_{hubName.ToLower()}";
         }
 
-        public string GenerateClientAccessToken(string hubName = null, IEnumerable<Claim> claims = null, TimeSpan? lifetime = null)
+        public async Task<string> GenerateClientAccessToken(string hubName = null, IEnumerable<Claim> claims = null, TimeSpan? lifetime = null)
         {
             var audience = $"{_endpoint}/{ClientPath}";
+            await _accessKey.AuthorizedTask;
```

Comment:
could `_accesskey` be null? add some defensive code in ctor?

Reply:
I think this should be promised by `Endpoint`? I could make sure that every `Endpoint` has a non-null **AccessKey**.

Reply:
we can have some defensive code for this class